pkg/util/homedir/homedir.go

```diff
@@ -23,7 +23,7 @@ import (
 
 // HomeDir returns the home directory for the current user
 func HomeDir() string {
-	if runtime.GOOS == "windows" {
+	if runtime.GOOS == "windows"; len(os.Getenv("HOME")) <= 0) {
```

Comment:
I thought that I could do something similar as the homedir.go, but then I noticed that 
RecommendedHomeFile is also used [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/client/unversioned/clientcmd/loader.go#L134-L149)

I am not entirely sure if this won't break anything further down the line.

Could you please have a look and let me know if you could recommend something else?
~~~Go
if goruntime.GOOS == "windows" {
    if home := os.Getenv("HOME"); len(home) > 0 {
        if _, err := os.Stat(home); err = nill {
            migrationRules[oldRecommendedWindowsHomeFile] = RecommendedHomeFile
        }
    } else {
        migrationRules[RecommendedHomeFile] = oldRecommendedWindowsHomeFile
    }
}
~~~